src/routes/messages/mappers/message-mapper.ts

```diff
@@ -0,0 +1,75 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { MessageConfirmation as DomainMessageConfirmation } from '../../../domain/messages/entities/message-confirmation.entity';
+import { Message as DomainMessage } from '../../../domain/messages/entities/message.entity';
+import { SafeAppsRepository } from '../../../domain/safe-apps/safe-apps.repository';
+import { ISafeAppsRepository } from '../../../domain/safe-apps/safe-apps.repository.interface';
+import { Safe } from '../../../domain/safe/entities/safe.entity';
+import { AddressInfoHelper } from '../../common/address-info/address-info.helper';
+import { MessageConfirmation } from '../entities/message-confirmation.entity';
+import { Message, MessageStatus } from '../entities/message.entity';
+
+@Injectable()
+export class MessageMapper {
+  constructor(
+    @Inject(ISafeAppsRepository)
+    private readonly safeAppsRepository: SafeAppsRepository,
+    private readonly addressInfoHelper: AddressInfoHelper,
+  ) {}
+
+  async mapMessage(
+    chainId: string,
+    message: DomainMessage,
+    safe: Safe,
+  ): Promise<Message> {
+    const safeApp = message.safeAppId
+      ? await this.safeAppsRepository.getSafeAppById(chainId, message.safeAppId)
+      : null;
+    const domainMessageConfirmations = message.confirmations ?? [];
+    const status =
+      domainMessageConfirmations.length >= safe.threshold
+        ? MessageStatus.Confirmed
+        : MessageStatus.NeedsConfirmation;
+    const proposedBy = await this.addressInfoHelper.getOrDefault(
+      chainId,
+      message.proposedBy,
+    );
+    const confirmations = await this.mapConfirmations(
+      chainId,
+      message.confirmations ?? [],
```

Comment:
We should always have a non-nullable array for confirmations right?

Reply:
According to the [Transaction Service specification](https://safe-transaction-mainnet.safe.global/), `confirmations` in `SafeMessageResponse` is optional, so I assumed this could be null (I defined this field as `nullable` in `message.schema.ts` as well), maybe I'm missing something here? 🙂

Reply:
> According to the [Transaction Service specification](https://safe-transaction-mainnet.safe.global/), `confirmations` in `SafeMessageResponse` is optional, so I assumed this could be null (I defined this field as `nullable` in `message.schema.ts` as well), maybe I'm missing something here? 🙂

The schema is automatically generated and it's not always correct. It says that `confirmations` is a `string` too but it's actually an array.

See https://github.com/safe-global/safe-transaction-service/blob/4e0d981c20c59f6524aa3a3032f0b07ff7b991f4/safe_transaction_service/safe_messages/serializers.py#L163-L172 (serializer uses `many=True` and is a required field)

Also the current implementation of the CGW: https://github.com/safe-global/safe-client-gateway/blob/12c8e08370218030a974613ecb733d8ca1398f33/src/routes/messages/backend_models.rs#L34

Reply:
Got it, fixed in [dc207bc](https://github.com/5afe/safe-client-gateway-nest/pull/275/commits/dc207bcc65410f71d4e6713e9bff1bd77f6d3b83)